src/docfx/build/dependency/DependencyResolver.cs

```diff
@@ -145,7 +146,16 @@ public string GetRelativeUrl(Document fileRelativeTo, Document file)
             // Cannot resolve the file, leave href as is
             if (file is null)
             {
-                return (error, href, fragment, hrefType, null);
+                if (string.IsNullOrEmpty(pathToDocset))
+                {
+                    return (error, href, fragment, hrefType, null);
+                }
+
+                (error, file) = TryResolveResourceFromHistory(_gitCommitProvider, relativeTo.Docset, pathToDocset);
```

Comment:
consider conslidating `TryResolveResourceFromHistory` and `TryResolveContentFromHistory` logic inside `TryResolveFile`. Logically, `ResolveContent` and `ResolveLink` boths build on top of `ResolveFile`, `ResolveContent` just takes the result of `ResolveFile` and read its content, `ResolveLink` just takes the result of `ResolveFile` and calculate url.

Reply:
the result `content` is the key point, the file resolved from history can't read `content` directly from `document`, so that's the reason why we put it outside of resolveFile. And for perf consideration, we don't want to read content for deleted resource.